paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/CustomerSessionCustomerSheetTest.kt

```diff
@@ -0,0 +1,219 @@
+                )
+            )
+        )
+
+        page.clickSaveButton()
+        page.clickConfirmButton()
+    }
+
+    @Test
+    fun testSuccessfulCardSaveWithCardBrandChoice() = runCustomerSheetTest(
+        networkRule = networkRule,
+        integrationType = integrationType,
+        customerSheetTestType = CustomerSheetTestType.CustomerSession,
+        resultCallback = { result ->
+            verifySelected(
+                expectedLast4 = "1001",
+                expectedBrand = CardBrand.CartesBancaires,
+                result = result,
+            )
+        }
+    ) { context ->
+        enqueueElementsSession(
+            cards = listOf(),
+            isCbcEligible = true,
+        )
+
+        context.presentCustomerSheet()
+
+        /*
+         * This card is overridden to use a test card compatible with CbcTestCardDelegate to skip
+         * checking card account ranges network operation which run only if account ranges aren't
+         * stores in memory.
+         */
+        page.fillOutCardDetails(
+            cardNumber = TEST_CBC_CARD_NUMBER
+        )
+        page.changeCardBrandChoice()
+
+        enqueueCbcPaymentMethodCreation()
+        enqueueSetupIntentRetrieval()
+        enqueueSetupIntentConfirmation()
+
+        enqueueElementsSession(
+            cards = listOf(
+                PaymentMethodFactory.card(id = "pm_12345").update(
+                    last4 = "1001",
+                    addCbcNetworks = true,
+                    brand = CardBrand.CartesBancaires,
+                )
+            ),
+            isCbcEligible = true,
+        )
+
+        page.clickSaveButton()
+        page.clickConfirmButton()
+    }
+
+    private fun enqueueElementsSession(
+        cards: List<PaymentMethod>,
+        isCbcEligible: Boolean = false,
+    ) {
+        val cardsArray = JSONArray()
+
+        cards.forEach { card ->
+            cardsArray.put(PaymentMethodFactory.convertCardToJson(card))
+        }
+
+        networkRule.enqueue(
+            host("api.stripe.com"),
+            method("GET"),
+            path("/v1/elements/sessions"),
+        ) { response ->
+            response.testBodyFromFile(
+                filename = "elements-sessions-requires_pm_with_cs.json",
+                replacements = listOf(
+                    ResponseReplacement(
+                        original = "[PAYMENT_METHODS_HERE]",
+                        new = cardsArray.toString(2),
+                    ),
+                    ResponseReplacement(
+                        original = "CARD_BRAND_CHOICE_ELIGIBILITY",
+                        new = isCbcEligible.toString(),
+                    ),
+                ),
+            )
+        }
+    }
+
+    private fun enqueuePaymentMethodCreation() {
+        networkRule.enqueue(
+            createPaymentMethodsRequest(),
+            cardDetailsParams(),
+            billingDetailsParams(),
+            bodyPart("allow_redisplay", "always"),
```

Comment:
This is the big piece of this test, ensuring that payment method creation always sets this to `always`